client/pkg/transport/listener.go

```diff
@@ -377,18 +381,32 @@ func (info TLSInfo) baseConfig() (*tls.Config, error) {
 	// Client certificates may be verified by either an exact match on the CN,
 	// or a more general check of the CN and SANs.
 	var verifyCertificate func(*x509.Certificate) bool
-	if info.AllowedCN != "" {
-		if info.AllowedHostname != "" {
-			return nil, fmt.Errorf("AllowedCN and AllowedHostname are mutually exclusive (cn=%q, hostname=%q)", info.AllowedCN, info.AllowedHostname)
+	var definedRestrictions int
+	for _, restriction := range []string{info.AllowedCN, info.AllowedHostname, info.AllowedURI} {
+		if restriction != "" {
+			definedRestrictions++
```

Comment:
How about returning here if `definedRestrictions > 1`, instead of having a case in the switch statement (line 391)?